MM2RandoLib/Randomizers/RText.cs

```diff
@@ -279,78 +297,88 @@ public void Randomize(Patch p, Random r)
 
             // Write Robot Master damage table
             StringBuilder sb;
-            for (int i = 0; i < txtRobos.Length; i++)
+            for (Int32 i = 0; i < txtRobos.Length; i++)
             {
                 sb = new StringBuilder();
 
                 // Since weaknesses are for the "Room", and the room bosses were shuffled,
                 // obtain the weakness for the boss at this room
                 // TODO: Optimize this mess; when the bossroom is shuffled it should save
                 // a mapping that could be reused here.
-                int newIndex = 0;
-                for (int m = 0; m < RandomMM2.randomBossInBossRoom.Components.Count; m++)
+                Int32 newIndex = 0;
+                for (Int32 m = 0; m < RandomMM2.randomBossInBossRoom.Components.Count; m++)
                 {
-                    var room = RandomMM2.randomBossInBossRoom.Components[m];
+                    RBossRoom.BossRoomRandomComponent room = RandomMM2.randomBossInBossRoom.Components[m];
+
                     if (room.OriginalBossIndex == i)
                     {
                         newIndex = m;
                         break;
                     }
                 }
 
-                for (int j = 0; j < 9; j++)
+                for (Int32 j = 0; j < 9; j++)
                 {
-                    int dmg = RWeaknesses.BotWeaknesses[newIndex, j];
-                    sb.Append($"{GetBossWeaknessDamageChar(dmg)} ");
+                    Int32 dmg = RWeaknesses.BotWeaknesses[newIndex, j];
+                    sb.Append($"{RText.GetBossWeaknessDamageChar(dmg)} ");
                 }
 
-                string rowString = sb.ToString().Trim();
-                for (int j = 0; j < rowString.Length; j++)
+                String rowString = sb.ToString().Trim();
```

Comment:
in C#, what is the difference between a `string` and a `String`?

Reply:
Maybe the latter works with inheritance which is what allows you to provide the string extension?

Reply:
There is no difference.  I prefer to use the Object type instead of the built-in type for consistency.